### PR TITLE
Rust bindings for ittapi

### PR DESCRIPTION
**Pull request by jlb6740**
When built this uses bindgen to generate rust bindings for ittapi.
Code is intended to be published at crates.io.

### CODE REVIEW

**Comment by jlb6740** [comment]
@ekovanova @dp7 
Hi .. This add rust bindings for ittapi. If published to crates.io it would allow someone who uses rust to add the dependency ittapi-rs to their project to link in ittapi. I was going to create a new project for this but that is probably not necessary. This patch is somewhat analogous to the fortran shim here: https://github.com/intel/ittapi/blob/master/include/ittnotify.f90. The only difference is that it would be prudent to run "cargo publish" on the code here so that it will be published at https://crates.io/ when there is an api update. This of course can be setup to be done automatically.

File: ittapi-rs/build.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#![allow(unused_imports)]
+use cmake;
+use cmake::Config;
+use std::env;
+use std::path::PathBuf;
+extern crate bindgen;
+
+#[cfg(target_os = "windows")]
+fn main() {}
```
Comment (ekovanova) [inline ittapi-rs/build.rs]: Do you plan to add support for Windows?

File: ittapi-rs/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "ittapi-rs"
+version = "0.1.0"
+authors = ["Johnnie Birch <45402135+jlb6740@users.noreply.github.com>"]
+edition = "2018"
```
[the remark this answers is not in the document]
Comment (jlb6740) [reply ittapi-rs/Cargo.toml]: Hi .. yes will update. Also here I think the version should be consistent with/reflect the version used for the ittapi. @ekovanova Where can I find that?

File: ittapi-rs/build.rs (hunk 1 above)
Comment (jlb6740) [reply ittapi-rs/build.rs]: I am not setup to test that so I was going to leave that support out of this initial push. Definitely would be ideal though.

File: ittapi-rs/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlb6740) [reply ittapi-rs/Cargo.toml]: @ekovanova Actually researching this I think 2018 is correct. It is the default set by cargo to the latest available edition. The item speaks to the edition of rust used not to some sort of timestamp of the creation of the package itself.

File: ittapi-rs/build.rs (hunk 1 above)
Comment (jlb6740) [reply ittapi-rs/build.rs]: Will work on for future edition and not lack of current support in the README on next push.

File: ittapi-rs/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekovanova) [reply ittapi-rs/Cargo.toml]: > Hi .. yes will update. Also here I think the version should be consistent with/reflect the version used for the ittapi. @ekovanova Where can I find that?

I'll provide ittapi static lib with the correct version.

**Pull request merged**